lib/cc/engine/analyzers/file_thread_pool.rb

```diff
@@ -14,11 +14,13 @@ def initialize(files, concurrency: DEFAULT_CONCURRENCY)
 
         def run(&block)
           queue = build_queue
+          lock = Mutex.new
 
           @workers = thread_count.times.map do
             Thread.new do
-              while !queue.empty? && (item = queue.pop(true))
-                yield item
+              until queue.empty?
+                item = lock.synchronize { queue.pop(true) unless queue.empty? }
+                yield item if item
```

Comment:
WDYT of a small utility method:

```rb
# yield item while (item = next_item(queue))

def next_item(queue)
  lock = Mutex.new
  lock.synchronize { !queue.empty? && queue.pop(true) }
end
```

Other option would be rescue the non-block-empty-queue-popped exception (assuming it's sufficiently specific), and replace it with a `nil`.